### PR TITLE
Update celo-bls-go to v0.3.3

### PR DESCRIPTION
**Pull request by mstraka100**
### Description

Updates the `celo-bls-go` dependency to `v0.3.3`. This adds several new architectures that the dependency can be compiled to, including that used by MacBook M1 chips.

### Tested

CI tests

### Backwards compatibility

The functionality of celo-bls-go has not changed with the update. No architectures were removed.

### CODE REVIEW

**Comment by piersy** [comment]
Coverage from tests in `./e2e_test/...` for `./consensus/istanbul/...` at commit 0f304caffac0a3b7971253af098e7593edea791f
<details><summary>coverage:  50.6% of statements across all listed packages</summary><br>
<pre>
coverage:  64.0% of statements in consensus/istanbul
coverage:  42.7% of statements in consensus/istanbul/announce
coverage:  55.7% of statements in consensus/istanbul/backend
coverage:   0.0% of statements in consensus/istanbul/backend/backendtest
coverage:  24.3% of statements in consensus/istanbul/backend/internal/replica
coverage:  65.8% of statements in consensus/istanbul/core
coverage:  50.0% of statements in consensus/istanbul/db
coverage:   0.0% of statements in consensus/istanbul/proxy
coverage:  64.4% of statements in consensus/istanbul/uptime
coverage:  51.8% of statements in consensus/istanbul/validator
coverage:  79.2% of statements in consensus/istanbul/validator/random
</pre>
CommentID: 29c6c9cdca
</details>

**Comment by palango** [comment]
Builds fine for me.

File: go.sum [hunk 1]
```diff
@@ -123,6 +123,20 @@ github.com/c-bata/go-prompt v0.2.2 h1:uyKRz6Z6DUyj49QVijyM339UJV9yhbr70gESwbNU3e
 github.com/c-bata/go-prompt v0.2.2/go.mod h1:VzqtzE2ksDBcdln8G7mk2RX9QyGjH+OVqOCSiVIqS34=
 github.com/celo-org/celo-bls-go v0.2.4 h1:V1y92kM5IRJWQZ6DCwqiKLW7swmUA5y/dPJ9YbU4HfA=
 github.com/celo-org/celo-bls-go v0.2.4/go.mod h1:eXUCLXu5F1yfd3M+3VaUk5ZUXaA0sLK2rWdLC1Cfaqo=
```
[the remark this answers is not in the document]
Comment (mstraka100) [reply go.sum:125]: Should they be? They weren't automatically so I assumed it was some kind of Go idiom to leave in commit locks for previous dependency versions

[the remark this answers is not in the document]
Comment (palango) [reply go.sum:125]: I was asking out of interest as I'm also new to Go. I don't see a reason why it should stay, so I hoped you have an explanation for me.

**Review by palango — APPROVED** [review]
The open question doesn't block this.

**Review by piersy — CHANGES_REQUESTED** [review]
I've submitted a pr that simply tidies the go modules, please check that - https://github.com/celo-org/celo-blockchain/pull/1959

File: go.sum (hunk 1 above)
[the remark this answers is not in the document]
Comment (piersy) [reply go.sum:125]: The go commands can easily automatically add missing dependencies but in order to remove dependencies you need to run a command that takes into account the whole scope of the project, and that would be `go mod tidy`.

See [here](https://go.dev/blog/using-go-modules#TOC_7.) for a more detailed description.

Also I raised a PR doing the tidy here - https://github.com/celo-org/celo-blockchain/pull/1959

**Review by piersy — APPROVED** [review]
Looking good now!

**Pull request merged**